src/content/language/dot-shorthand.md

```diff
@@ -0,0 +1,328 @@
+```
+
+
+## Advanced usage
+
+### Chaining
+While you can chain operations like method calls or 
+property accesses onto a dot shorthand, the entire expression 
+is validated against the context type.
+
+The compiler first uses the context to determine what the 
+dot shorthand resolves to. Any subsequent operations in the 
+chain must return a value that matches that same initial 
+context type.
+
+
+<?code-excerpt "language/lib/classes/shorthand.dart (chain)"?>
+```dart
+// .fromCharCode(72) resolves to the String "H",
+// then the instance method .toLowerCase() is called on that String.
+String lowerH = .fromCharCode(72).toLowerCase(); // Instead of String.fromCharCode(72).toLowerCase()
+
+print(lowerH); // Output: h
+```
+
+### Equality operators (`==` and `!=`)
+The `==` and `!=` operators have a special rule for dot shorthands. 
+When dot shorthand syntax is used directly on the right-hand side (RHS) 
+of an equality check, Dart uses the static type of the 
+left-hand side to determine the class or enum for the shorthand.
+
+For instance, in an expression like `myColor == .green`, 
+the type of the variable `myColor` is used as the context. 
+This means the compiler interprets `.green` as `Color.green`.
+
+<?code-excerpt "language/lib/classes/shorthand.dart (allowedequality)"?>
+```dart
+enum Color { red, green, blue }
+
+void allowedExamples() {
+  Color myColor = Color.red;
+  bool condition = true;
+
+  // OK: `myColor` is a `Color`, so `.green` is inferred as `Color.green`.
+  if (myColor == .green) {
+    print('The color is green.');
+  }
+
+  // OK: Works with `!=` as well.
+  if (myColor != .blue) {
+    print('The color is not blue.');
+  }
+
+  // OK: The context for the ternary is the variable `inferredColor`
+  // being assigned to, which has a type of `Color`.
+  Color inferredColor = condition ? .green : .blue;
+  print('Inferred color is $inferredColor');
+}
+```
+The dot shorthand must be on the right-hand side of the `==` 
+or `!=` operator. Comparing against a more complex expression, 
+like a ternary, is also not allowed.
+
+<?code-excerpt "language/lib/classes/shorthand.dart (notallowedequality)"?>
+```dart
+enum Color { red, green, blue }
+
+void notAllowedExamples() {
+  Color myColor = Color.red;
+  bool condition = true;
+
+  // ERROR: The shorthand must be on the right side of `==`.
+  // Dart's `==` operator is not symmetric for this feature.
+  if (.red == myColor) {
+    print('This will not compile.');
+  }
+
+  // ERROR: The right-hand side is a complex expression (a ternary),
+  // which is not a valid target for shorthand in a comparison.
+  if (myColor == (condition ? .green : .blue)) {
+    print('This will not compile.');
+  }
+
+  // ERROR: The type context is lost by casting `myColor` to `Object`.
+  // The compiler no longer knows that `.green` should refer to `Color.green`.
+  if ((myColor as Object) == .green) {
+    print('This will not compile.');
+  }
+}
+```
+
+## Best practices 
```

Comment:
I wonder if we should have best practices at all? I chatted with Bob a bit about it and it seems like we typically don't prescribe anything and just allow people to use it. Maybe once we've seen how people use it, we'd come back and update with more guidance. I'm not sure if we should be telling people to avoid using it in any sort of way right now (even though Loic's dogfooding had specific recs)

Wonder if Amanda has any guidance on this?